app/models/user.rb

```diff
@@ -623,63 +628,30 @@ def flipper_id
 
   private
 
-  # rubocop:disable Metrics/BlockLength
-  def migrate_users_fields_to_users_settings(users_setting_record)
+  def sync_relevant_profile_fields_to_user_settings_table(users_setting_record)
+    PROFILE_FIELDS_TO_MIGRATE_TO_USERS_SETTINGS_TABLE.each do |field|
+      users_setting_record.update(field => profile.data[field]) if profile.data[field].present?
+    end
+  end
+
+  def migrate_users_and_profile_fields_to_users_settings(users_setting_record)
     USER_FIELDS_TO_MIGRATE_TO_USERS_SETTINGS_TABLE.each do |field|
-      case field
-      when "config_font"
-        config_font_enums = {
-          default: 0,
-          comic_sans: 1,
-          monospace: 2,
-          open_dyslexic: 3,
-          sans_serif: 4,
-          serif: 5
-        }
-        users_setting_record.update(field => config_font_enums[public_send(field).to_sym])
-      when "config_navbar"
-        config_navbar_enums = {
-          default: 0,
-          static: 1
-        }
-        users_setting_record.update(field => config_navbar_enums[public_send(field).to_sym])
-      when "config_theme"
-        config_theme_enums = {
-          default: 0,
-          minimal_light_theme: 1,
-          night_theme: 2,
-          pink_theme: 3,
-          ten_x_hacker_theme: 4
-        }
-        users_setting_record.update(field => config_theme_enums[public_send(field).to_sym])
-      when "editor_version"
-        config_editor_enums = {
-          v2: 0,
-          v1: 1
-        }
-        users_setting_record.update(field => config_editor_enums[public_send(field).to_sym])
-      when "inbox_type"
-        config_inbox_enums = {
-          private: 0,
-          open: 1
-        }
-        users_setting_record.update(field => config_inbox_enums[public_send(field).to_sym])
+      if USER_SETTINGS_ENUM_FIELDS.include?(field)
+        field_enums = Users::Setting.defined_enums[field]
+        users_setting_record.update(field => field_enums[public_send(field).to_sym])
       else
         users_setting_record.update(field => public_send(field))
       end
     end
 
-    PROFILE_FIELDS_TO_MIGRATE_TO_USERS_SETTINGS_TABLE.each do |field|
-      users_setting_record.update(field => profile.public_send(field)) if profile.public_send(field).present?
-    end
+    sync_relevant_profile_fields_to_user_settings_table(users_setting_record) if profile
```

Comment:
I'm conflicted about this line: while I understand that we don't want to raise exceptions while syncing I also don't think we should silently ignore the fact that a user is missing their profile because that shouldn't happen in the first place.